crypto/bls_core.c

```diff
@@ -22,24 +22,16 @@ int get_sk_len() {
 // checks an input scalar a satisfies 0 < a < r
 // where (r) is the order of G1/G2
 int check_membership_Zr_star(const bn_t a){
-    int ret; // return value
-    bn_t r;
-    bn_new(r); 
-    g2_get_ord(r);
-    if (bn_cmp(a,r) != RLC_LT || bn_cmp_dig(a, 0) != RLC_GT) ret = INVALID; 
-    else ret = VALID;
-    bn_free(r);
-    return ret;
+    if (bn_cmp(a, &core_get()->ep_r) != RLC_LT || bn_cmp_dig(a, 0) != RLC_GT) {
+        return INVALID; 
+    }
+    return VALID;
 }
 
-// checks if input point s is on the curve E1
-// and is in the subgroup G1. 
+// Checks if input point s is in the subgroup G1. 
```

Comment:
If I read that correctly (haven't worked with C for quite some time 😅 ), the function input is called `p` (?) 
```suggestion
// Checks if input point p is in the subgroup G1. 
```

Reply:
You are right, good point :)